src/bindings/generator/src/dbus_typing.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+from typing import Tuple
+import dasbus.typing as DBusTypes
+
+basic_types = ["y", "b", "n", "q", "i", "u", "x", "t", "d", "s", "o", "g", "h", "v"]
+
+
+def map_basic_type_to_pytype(t: str) -> str:
+    if t == "y":
+        return DBusTypes.get_type_name(DBusTypes.Byte)
+    elif t == "b":
+        return DBusTypes.get_type_name(DBusTypes.Bool)
+    elif t == "n":
+        return DBusTypes.get_type_name(DBusTypes.Int16)
+    elif t == "q":
+        return DBusTypes.get_type_name(DBusTypes.UInt16)
+    elif t == "i":
+        return DBusTypes.get_type_name(DBusTypes.Int32)
+    elif t == "u":
+        return DBusTypes.get_type_name(DBusTypes.UInt32)
+    elif t == "x":
+        return DBusTypes.get_type_name(DBusTypes.Int16)
+    elif t == "t":
+        return DBusTypes.get_type_name(DBusTypes.UInt64)
+    elif t == "d":
+        return DBusTypes.get_type_name(DBusTypes.Double)
+    elif t == "s":
+        return DBusTypes.get_type_name(DBusTypes.Str)
+    elif t == "o":
+        return DBusTypes.get_type_name(DBusTypes.ObjPath)
+    elif t == "g":
+        return DBusTypes.get_type_name(DBusTypes.Str)
+    elif t == "h":
+        return DBusTypes.get_type_name(DBusTypes.UnixFD)
+    elif t == "v":
+        return DBusTypes.get_type_name(DBusTypes.Variant)
+
+    raise Exception(f"Unknown basic type '{t}'")
+
+
+def map_array_type_to_pytype(t: str) -> Tuple[str, int]:
```

Comment:
Would help a docstring explaining what this return tuple does in the end. I understood it's array -> pytype but nothing more than that. :(

Reply:
Fair point - refactored it and added inline comments that, hopefully, help understanding what the code does.